pilot/cmd/pilot-agent/app/cmd.go

```diff
@@ -173,6 +173,7 @@ func newProxyCommand() *cobra.Command {
 				return err
 			}
 			wait()
+			agent.Close()
```

Comment:
do we want defer so it closes on errors as well?

Reply:
Makes sense. Let me change